lib/src/outputs/advanced_file_output.dart

```diff
@@ -0,0 +1,194 @@
+  ///
+  /// [maxDelay] describes the maximum amount of time before the buffer has to be
+  /// written to the file.
+  ///
+  /// Any log levels that are specified in [writeImmediately] trigger an immediate
+  /// flush to the disk ([Level.warning], [Level.error] and [Level.fatal] by default).
+  ///
+  /// [path] is either treated as directory for rotating or as target file name,
+  /// depending on [maxFileSizeKB].
+  AdvancedFileOutput({
+    required String path,
+    bool overrideExisting = false,
+    Encoding encoding = utf8,
+    List<Level>? writeImmediately,
+    Duration maxDelay = const Duration(seconds: 2),
+    int maxBufferSize = 2000,
+    int maxFileSizeKB = 1024,
+    String latestFileName = 'latest.log',
+    String Function(DateTime timestamp)? fileNameFormatter,
+  })  : _path = path,
+        _overrideExisting = overrideExisting,
+        _encoding = encoding,
+        _maxDelay = maxDelay,
+        _maxFileSizeKB = maxFileSizeKB,
+        _maxBufferSize = maxBufferSize,
+        _fileNameFormatter = fileNameFormatter ?? _defaultFileNameFormat,
+        _writeImmediately = writeImmediately ??
+            [
+              Level.error,
+              Level.fatal,
+              Level.warning,
+              // ignore: deprecated_member_use_from_same_package
+              Level.wtf,
+            ],
+        _file = maxFileSizeKB > 0 ? File('$path/$latestFileName') : File(path);
+
+  /// Logs directory path by default, particular log file path if [_maxFileSizeKB] is 0.
+  final String _path;
+
+  final bool _overrideExisting;
+  final Encoding _encoding;
+
+  final List<Level> _writeImmediately;
+  final Duration _maxDelay;
+  final int _maxFileSizeKB;
+  final int _maxBufferSize;
+  final String Function(DateTime timestamp) _fileNameFormatter;
+
+  final File _file;
+  IOSink? _sink;
+  Timer? _bufferFlushTimer;
+  Timer? _targetFileUpdater;
+
+  final List<OutputEvent> _buffer = [];
+
+  bool get _rotatingFilesMode => _maxFileSizeKB > 0;
+
+  /// Formats the file with a full date string.
+  ///
+  /// Example:
+  /// * `2024-01-01-10-05-02-123.log`
+  static String _defaultFileNameFormat(DateTime t) {
+    return '${t.year}-${t.month.toDigits(2)}-${t.day.toDigits(2)}'
+        '-${t.hour.toDigits(2)}-${t.minute.toDigits(2)}-${t.second.toDigits(2)}'
+        '-${t.millisecond.toDigits(3)}.log';
+  }
+
+  @override
+  Future<void> init() async {
+    if (_rotatingFilesMode) {
+      final dir = Directory(_path);
+      // We use sync directory check to avoid losing potential initial boot logs
+      // in early crash scenarios.
+      if (!dir.existsSync()) {
+        dir.createSync(recursive: true);
+      }
+
+      _targetFileUpdater = Timer.periodic(
+        const Duration(minutes: 1),
+        (_) => _updateTargetFile(),
+      );
+    }
+
+    _bufferFlushTimer = Timer.periodic(_maxDelay, (_) => _flushBuffer());
+    await _openSink();
+    await _updateTargetFile(); // Run first check without waiting for timer tick
```

Comment:
I would suggest we also skip this call, if `_maxFileSizeKB` is not a valid size.

```suggestion
    if (_rotatingFilesMode) {
      await _updateTargetFile(); // Run first check without waiting for timer tick
    }
```